src/auth.zig

```diff
@@ -107,6 +97,320 @@ test MTLS {
     // TODO there's likely a few more error states we should validate;
 }
 
+/// Default CookieAuth Helper uses Sha256 as the HMAC primitive.
+pub const Cookie = CookieAuth(Hmac.sha2.HmacSha256);
+
+pub const cookie_auth = struct {
+    /// Why are you using sha1?
+    pub const sha1 = CookieAuth(Hmac.HmacSha1);
+
+    pub const sha2 = struct {
+        pub const @"224" = CookieAuth(Hmac.sha2.HmacSha224);
+        pub const @"256" = CookieAuth(Hmac.sha2.HmacSha256);
+        pub const @"384" = CookieAuth(Hmac.sha2.HmacSha384);
+        pub const @"512" = CookieAuth(Hmac.sha2.HmacSha512);
+    };
+};
+
+/// TODO document
+pub fn CookieAuth(HMAC: type) type {
```

Comment:
Should we go ahead and move the cookie auth and mtls implementation into their own file in the auth directory and just import them here?

Reply:
yes, I'll do that in the next PR